auto_nag/user_activity.py

```diff
@@ -0,0 +1,88 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import datetime, timedelta
+from enum import Enum
+
+from libmozdata.bugzilla import Bugzilla, BugzillaUser
+
+from auto_nag import utils
+
+DEFULT_ACTIVITY_WEEKS = 26
+
+
+class UserStatus(Enum):
+    UNDEFINED = 1
+    DISABLED = 2
+    INACTIVE = 3
+
+    def __str__(self) -> str:
+        return self.name
+
+
+class UserActivity:
+    def __init__(self, weeks_count=DEFULT_ACTIVITY_WEEKS) -> None:
+        self.weeks_count = weeks_count
+
+    def check_users(self, user_emails):
+        user_emails = set(user_emails)
+
+        none_users = {
+            user_email for user_email in user_emails if utils.is_no_assignee(user_email)
+        }
+        user_emails.difference_update(none_users)
+        status = {user_email: UserStatus.UNDEFINED for user_email in none_users}
+
+        disabled_users = self.get_disabled_users(user_emails)
+        user_emails.difference_update(disabled_users)
+        for user_email in disabled_users:
+            status[user_email] = UserStatus.DISABLED
+
+        for user_email in user_emails:
+            if self.is_inactive_user(user_email):
```

Comment:
We could avoid this when we already detected that the user is disabled

Reply:
Disabled users already excluded from the `user_emails` list in L38:

https://github.com/mozilla/relman-auto-nag/blob/7f331d281eff81dbe182892cea180be523a2d11b/auto_nag/user_activity.py#L38

Reply:
Oh, I hadn't noticed...